fastlane/lib/fastlane/actions/download_dsyms.rb

```diff
@@ -101,7 +101,7 @@ def self.run(params)
 
           if after_uploaded_date && after_uploaded_date >= uploaded_date
             UI.verbose("Upload date #{after_uploaded_date} not reached: #{uploaded_date}")
-            break
+            next
```

Comment:
Using next here doesn't really make sense to me. I tried to reason about it in my commit message in dac7592c98e0bb84028b5b694558182c8ae49dbd

Since we know the results are sorted by uploadedDate, I think it should be safe to break out of the loop when we're past the expected date already.

Reply:
@tmm1  Yeah yeah, that does seem okay but I don't _fully_ trust they will be returned in the right order 😬 I trust the official ASC API but we also use the proxy endpoint for Apple ID auth which is the one I don't really trust. So I _think_ I'd feel safer here doing a next just to make sure all the correct ones are found. The slowest part was the API request so this _shouldn't_ increase things too much 🤞 

I hope my insane uneasiness makes sense 😛

Reply:
Fair enough :boat: